velox/exec/Driver.h

```diff
@@ -244,6 +244,43 @@ struct DriverCtx {
   std::optional<common::SpillConfig> makeSpillConfig(int32_t operatorId) const;
 };
 
+/// Enum for operator methods we want to track.
+enum class OperatorMethod {
+  kNone,
+  kIsBlocked,
+  kNeedsInput,
+  kGetOutput,
+  kAddInput,
+  kNoMoreInput
+};
+
+/// Structure holds the information about the current operator call the driver
+/// is in. Can be used to detect deadlocks and otherwise blocked calls.
+/// If timeStartMs is zero, then we aren't in an operator call.
+struct OpCallStatus {
+  /// Time (ms) when the operator call started.
+  std::atomic_size_t timeStartMs{0};
+  /// Id of the operator, method of which is currently running. It is index into
+  /// the vector of Driver's operators.
+  std::atomic_int32_t opId{0};
+  /// Method of the operator, which is currently running.
+  std::atomic<OperatorMethod> method{OperatorMethod::kNone};
+
+  OpCallStatus() {}
+  OpCallStatus(const OpCallStatus& other);
```

Comment:
Why we have to make it copiable?  Can we keep a `unique_ptr<OpCallStatus>` so we can forget about copying it?

Reply:
@Yuhta 
Why not?
What is the benefit of one indirection (pointer is a indirection)?

But you gave me an interesting thought: how about a OpCallStatusRaw which contains integers, not atomics? And instead of copying we just return that raw structure.

Reply:
The copy is not very well defined in that way.  The `OpCallStatusRaw` approach is much better.